dotnet/src/Microsoft.Agents.AI.DevUI/EntitiesApiExtensions.cs

```diff
@@ -0,0 +1,242 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Text.Json;
+
+using Microsoft.Agents.AI.DevUI.Entities;
+using Microsoft.Agents.AI.Hosting;
+
+namespace Microsoft.Agents.AI.DevUI;
+
+/// <summary>
+/// Provides extension methods for mapping entity discovery and management endpoints to an <see cref="IEndpointRouteBuilder"/>.
+/// </summary>
+public static class EntitiesApiExtensions
+{
+    /// <summary>
+    /// Maps HTTP API endpoints for entity discovery and management.
+    /// </summary>
+    /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/> to add the routes to.</param>
+    /// <returns>The <see cref="IEndpointRouteBuilder"/> for method chaining.</returns>
+    /// <remarks>
+    /// This extension method registers the following endpoints:
+    /// <list type="bullet">
+    /// <item><description>GET /v1/entities - List all registered entities (agents and workflows)</description></item>
+    /// <item><description>GET /v1/entities/{entityId}/info - Get detailed information about a specific entity</description></item>
+    /// </list>
+    /// The endpoints are compatible with the Python DevUI frontend and automatically discover entities
+    /// from the registered <see cref="AgentCatalog"/> and <see cref="WorkflowCatalog"/> services.
+    /// </remarks>
+    public static IEndpointRouteBuilder MapEntities(this IEndpointRouteBuilder endpoints)
+    {
+        var group = endpoints.MapGroup("/v1/entities")
+            .WithTags("Entities");
+
+        // List all entities
+        group.MapGet("", ListEntitiesAsync)
+            .WithName("ListEntities")
+            .WithSummary("List all registered entities (agents and workflows)")
+            .Produces<DiscoveryResponse>(StatusCodes.Status200OK, contentType: "application/json");
+
+        // Get detailed entity information
+        group.MapGet("{entityId}/info", GetEntityInfoAsync)
+            .WithName("GetEntityInfo")
+            .WithSummary("Get detailed information about a specific entity")
+            .Produces<EntityInfo>(StatusCodes.Status200OK, contentType: "application/json")
+            .Produces(StatusCodes.Status404NotFound);
+
+        return endpoints;
+    }
+
+    private static async Task<IResult> ListEntitiesAsync(
+        AgentCatalog? agentCatalog,
+        WorkflowCatalog? workflowCatalog,
+        CancellationToken cancellationToken)
+    {
+        try
+        {
+            var entities = new List<EntityInfo>();
+
+            // Discover agents from the agent catalog
+            if (agentCatalog is not null)
+            {
+                await foreach (var agent in agentCatalog.GetAgentsAsync(cancellationToken).ConfigureAwait(false))
+                {
+                    if (agent.GetType().Name == "WorkflowHostAgent")
```

Comment:
This seems pretty fragile and that there's a decent chance the internal type will get renamed, no one will know to update this, and it'll just silently be skipped

Reply:
I recommend we add an API that allows the app to pass in agents that should be excluded from the devUI. In the prototype Luis had built up, he hid "internal" agents from the devui where a workflow wrapped around those agents, so that only the workflow agent would be shown and the others would not be. He did that by implementing a filter in his corresponding method for discovering the registered agents.